sound/squeezelite/Makefile

```diff
@@ -28,104 +28,185 @@ define Package/squeezelite/default
     TITLE:=Headless squeezebox emulator
     PROVIDES:=squeezelite
     URL:=https://github.com/ralph-irving/squeezelite
-    DEPENDS:= +alsa-lib +SQUEEZELITE_RESAMPLE:libsoxr
-    MENU:=1
+    DEPENDS:=+alsa-lib
 endef
 
 define Package/squeezelite-full
     $(call Package/squeezelite/default)
     TITLE+= (full)
-    DEPENDS+= +libflac +libvorbis +libmpg123 +libfaad2 \
-              +SQUEEZELITE_OPUS:libopusfile \
-              +SQUEEZELITE_SSL:libopenssl \
-              +SQUEEZELITE_WMA:libffmpeg-audio-dec
     VARIANT:=full
+    DEPENDS+= +libfaad2 +libffmpeg-audio-dec +libflac +libmpg123 \
+              +libopenssl +libopusfile +libsoxr +libvorbis
 endef
 
-define Package/squeezelite-mini
+define Package/squeezelite-dynamic
     $(call Package/squeezelite/default)
-    TITLE+= (minimal)
-    VARIANT:=mini
-endef
-
-define Package/squeezelite/config/default
-
-	config SQUEEZELITE_WMA
-	    bool "WMA/ALAC decode support"
-	    depends on BUILD_PATENTED
-	    help
-		Include WMA and ALAC decoding using ffmpeg
-	    default n
-
-	config SQUEEZELITE_RESAMPLE
-	    bool "Resample support"
-	    help
-		Include support for resampling using libsoxr
-	    default n
-
-	config SQUEEZELITE_DSD
-	    bool "DSD playback over PCM (DoP)"
-	    help
-		Include support for DSD over PCM for compatible DAC
-	    default n
-
-	config SQUEEZELITE_SSL
-	    bool "SSL/TLS support"
-	    help
-		Include SSL/TLS support for use with e.g. https media URLs
-	    default n
-
-	config SQUEEZELITE_OPUS
-	    bool "Opus codec support"
-	    help
-		Include Opus codec support
-	    default n
+    TITLE+= (dynamic)
+    VARIANT:=dynamic
+    DEPENDS+= +libmpg123
+    PKG_BUILD_DEPENDS:=faad2 ffmpeg flac libsoxr libvorbis \
```

Comment:
The dynamic variant still contains code to decode aac, alac, flac, ogg, opus, ... by using dynamically loaded libraries. The code needed to interface with the decode libraries is small, but it uses data structures defined in the decode libraries and that's why I added PKG_BUILD_DEPENDS so that header files are available when building the decode "stubs".

Reply:
Does PKG_BUILD_DEPENDS even work when added here?

Reply:
No, it didn't work, thanks! :-)

I moved it to the top of the file. I think the only negative side effect of this is that a slimmed down custom variant build will build dependencies that aren't needed.

Is that ok or do you see any other way of handling this?

Reply:
I find no issue with that.